### PR TITLE
Promote min ready sec sts beta

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
StatefulSet minReadySeconds is promoted to beta
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/pull/2824
```

### CODE REVIEW

**Comment by ravisantoshgudimetla** [comment]
cc @atiratree @soltysh

**Review by soltysh — APPROVED** [review]
Rebase and this is ready to go

**Review by atiratree — APPROVED** [review]

**Comment by smarterclayton** [comment]
Is there an STS minReadySeconds e2e?

**Comment by ravisantoshgudimetla** [comment]
@smarterclayton - Here is the test which exercises sts e2e

**Comment by smarterclayton** [comment]
/approve

Changes

/lgtm

based on other comments and review

**Comment by ravisantoshgudimetla** [comment]
``` Message: "Invalid value: 30741: provided port is already allocated",``` networking test failure which is unrelated

All integration tests passed, I think it's an issue with clean-up code.

/retest

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -1420,7 +1420,8 @@
         }
       },
       "required": [
-        "replicas"
+        "replicas",
+        "availableReplicas"
```
Comment (liggitt) [inline api/openapi-spec/swagger.json:1424]: adding a new field marked as required is not backwards compatible... this should be marked optional

Comment (alculquicondor) [reply api/openapi-spec/swagger.json:1424]: Can this be fixed reverted in the next patch version?

Comment (liggitt) [reply api/openapi-spec/swagger.json:1424]: yes, @ravisantoshgudimetla is opening a PR and will pick it back

**Pull request merged**